Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -245,72 +241,89 @@ extension InternalRouter where Self: Router {
         if isRerouting { return }
         isRerouting = true
         
-        getDirections(from: location, along: routeProgress) { [weak self] (session, result) in
-            self?.isRerouting = false
-            
-            guard case let .success(response) = result else {
-                return
-            }
-            guard let route = response.routes?.first else { return }
-            
-            self?.lastProactiveRerouteDate = nil
+        calculateRoutes(from: location, along: routeProgress) { [weak self] (session, result) in
+            guard let strongSelf = self else { return }
+            strongSelf.isRerouting = false
             
-            guard let firstLeg = route.legs.first, let firstStep = firstLeg.steps.first else {
+            switch result {
+            case .failure:
                 return
-            }
-            
-            let routeIsFaster = firstStep.expectedTravelTime >= RouteControllerMediumAlertInterval &&
-                currentUpcomingManeuver == firstLeg.steps[1] && route.expectedTravelTime <= 0.9 * durationRemaining
-            
-            if routeIsFaster {
-                var routeOptions: RouteOptions?
-                if case let .route(options) = response.options {
-                    routeOptions = options
+            case .success(let indexedResponse):
+                let response = indexedResponse.routeResponse
+                guard let route = response.routes?.first else { return }
+                
+                strongSelf.lastProactiveRerouteDate = nil
+                
+                guard let firstLeg = route.legs.first, let firstStep = firstLeg.steps.first else {
+                    return
                 }
                 
-                self?.indexedRouteResponse = .init(routeResponse: response, routeIndex: 0)
-                self?.setRoute(route, proactive: true, routeOptions: routeOptions)
+                let routeIsFaster = firstStep.expectedTravelTime >= RouteControllerMediumAlertInterval &&
+                    currentUpcomingManeuver == firstLeg.steps[1] && route.expectedTravelTime <= 0.9 * durationRemaining
+                
+                if routeIsFaster {
```

Comment:
maybe use `guard` here? the level of nesting is too deep imho
also, I want to add a linter on the length of the line in the near future

Reply:
I changed it back to a guard statement.